libraries/redshop/helper/product.php

```diff
@@ -5133,5 +5136,39 @@ public static function checkCompareProduct($productId)
         return '';
     }
 
+    /**
+     * @param   integer    $productId     Product ID
+     * @param   float    $productPrice  Product price
+     *
+     * @return  float
+     *
+     */
+    public function calculateProductApplyMassDiscount($productId, $productPrice)
+    {
+        $productId = (int)$productId;
+
+        if (!$productId) {
+            return '';
+        }
+        $product = Redshop\Product\Product::getProductById($productId);
+
+        $db    = JFactory::getDbo();
+        $query = $db->getQuery(true)
+            ->select('*')
+            ->from($db->qn('#__redshop_mass_discount'))
+            ->where(' FIND_IN_SET(' . (int)$productId . ',' . $db->quoteName('discount_product') . ')');
```

Comment:
@hoangvan149dhv It's totaly wrong if mass discount set for category or manufacturer, that case `discount_product` will be null.